From cff74c1efe7c4d6ecb0a7ccb9c8d86511b1aeac7 Mon Sep 17 00:00:00 2001
From: Felix Radensky <felix.r@variscite.com>
Date: Thu, 3 Dec 2020 18:10:24 +0200
Subject: [PATCH] usb: gadget: Fix ethernet gadget compilation failure

When CONFIG_USB_ETHER is defined but CONFIG_SPL_USB_ETHER is not
the SPL complation fails. This patch fixes the problem.

Signed-off-by: Nate Drude <nate.d@variscite.com>
(cherry picked from commit 651ecf057573e832e995db1d85da55f350415a68)
---
 drivers/usb/gadget/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/Makefile b/drivers/usb/gadget/Makefile
index d5d891b2053..805433b412e 100644
--- a/drivers/usb/gadget/Makefile
+++ b/drivers/usb/gadget/Makefile
@@ -33,7 +33,7 @@ obj-$(CONFIG_USB_FUNCTION_ROCKUSB) += f_rockusb.o
 obj-$(CONFIG_USB_FUNCTION_ACM)	+= f_acm.o
 endif
 endif
-ifdef CONFIG_USB_ETHER
+ifdef CONFIG_$(SPL_)USB_ETHER
 obj-y += ether.o
 obj-$(CONFIG_USB_ETH_RNDIS) += rndis.o
 obj-$(CONFIG_CI_UDC)	+= ci_udc.o
-- 
2.34.1

